wp-admin/js/widgets/media-image-widget.js

```diff
@@ -43,32 +43,78 @@
 		 * @returns {object} Props
 		 */
 		getSelectFrameProps: function getSelectFrameProps( mediaFrame ) {
-			var attachment, displaySettings, props;
+			var control = this,
+				state = mediaFrame.state(),
+				props = {};
+
+			if ( 'embed' === state.get( 'id' ) ) {
+				props = control._getEmbedProps( state.props.toJSON() );
+			} else {
+				props = control._getAttachmentProps( state.get( 'selection' ).first().toJSON() );
+			}
+
+			return props;
+		},
 
-			attachment = mediaFrame.state().get( 'selection' ).first().toJSON();
-			if ( _.isEmpty( attachment ) ) {
-				return {};
+		/**
+		 * Get the instance props from the media selection frame.
+		 *
+		 * @param {object} attachment Attachment object.
+		 * @returns {object} Props
+		 */
+		_getAttachmentProps: function _getAttachmentProps( attachment ) {
+			var props = {};
+
+			if ( ! _.isEmpty( attachment ) ) {
+				props = {
+					attachment_id: attachment.id,
+					alt: attachment.alt,
+					caption: attachment.caption,
+					image_classes: '',
+					image_title: '',
+					link_classes: '',
+					link_rel: '',
+					link_url: attachment.link,
+					link_target_blank: false,
+					link_type: attachment.link,
+					size: 'thumbnail',
+					url: attachment.sizes.thumbnail.url,
+					width: 0, // Reset.
+					height: 0 // Reset.
+				};
 			}
 
-			displaySettings = mediaFrame.content.get( '.attachments-browser' ).sidebar.get( 'display' ).model.toJSON();
-
-			props = {
-				attachment_id: attachment.id,
-				align: displaySettings.align,
-				alt: attachment.alt,
-				caption: attachment.caption,
-				image_classes: '',
-				image_title: '',
-				link_classes: '',
-				link_rel: '',
-				link_url: displaySettings.linkUrl,
-				link_target_blank: false,
-				link_type: displaySettings.link,
-				size: displaySettings.size,
-				url: attachment.sizes[ displaySettings.size ].url,
-				width: 0, // Reset.
-				height: 0 // Reset.
-			};
+			return props;
+		},
+
+		/**
+		 * Get the instance props from the media selection frame.
+		 *
+		 * @param {object} attachment Attachment object.
+		 * @returns {object} Props
+		 */
+		_getEmbedProps: function _getEmbedProps( attachment ) {
+			var props = {};
+
+			if ( ! _.isEmpty( attachment ) ) {
+				props = {
+					attachment_id: 0,
```

Comment:
I see. The `attachment.id` is retained in `ImageWidgetModel#_getAttachmentProps` but is not relevant here in `ImageWidgetModel#_getEmbedProps` since there is no underlying attachment.